io.github.fastrizwaan.WineCharm.yaml

```diff
@@ -0,0 +1,220 @@
+        sha256: 17abe02d043a253b68b47e3af69c9fc755b895db68fdc8811786125df564c6e0
+
+  - name: perl
+    no-autogen: true
+    config-opts:
+    - "-des"
+    - "-Duseshrplib"
+    build-options:
+      cflags: "-fPIC"
+      ldflags: "-fpic"
+    sources:
+    - type: archive
+      url: https://www.cpan.org/src/5.0/perl-5.40.2.tar.xz
+      sha256: 0551c717458e703ef7972307ab19385edfa231198d88998df74e12226abf563b
+    - type: script
+      dest-filename: configure
+      commands:
+      - exec ./configure.gnu $@
+    post-install:
+      - find /app/lib/perl5 -type f -exec chmod u+w {} \;
+  
+  - name: exiftool
+    buildsystem: simple
+    build-commands:
+      - perl Makefile.PL
+      - make
+      - make install 
+    sources:
+      - type: archive
+        url: https://exiftool.org/Image-ExifTool-13.30.tar.gz
+        sha256: 885afd06c4efcc60d1df703cc88ba7ddc3bb6fed854cfbaa9e6cd72adfbe8da9
+        
+  - name: desktop-file-utils
+    build-options:
+      no-debuginfo: true  
+    buildsystem: meson
+    sources:
+      - type: archive
+        url: https://www.freedesktop.org/software/desktop-file-utils/releases/desktop-file-utils-0.28.tar.xz
+        sha256: 4401d4e231d842c2de8242395a74a395ca468cd96f5f610d822df33594898a70
+
+  - name: psutil
+    buildsystem: simple
+    build-commands:
+      - pip3 install --prefix=/app psutil*.whl
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/19/74/f59e7e0d392bc1070e9a70e2f9190d652487ac115bb16e2eff6b22ad1d24/psutil-6.0.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+        sha256: 5fd9a97c8e94059b0ef54a7d4baf13b405011176c3b6ff257c247cae0d560ecd
+        only-arches:
+          - x86_64
+        x-checker-data:
+          type: pypi
+          name: psutil
+          packagetype: bdist_wheel
+
+  - name: pyyaml
+    buildsystem: simple
+    build-commands:
+      - pip3 install --prefix=/app PyYAML*.whl
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/b9/2b/614b4752f2e127db5cc206abc23a8c19678e92b23c3db30fc86ab731d3bd/PyYAML-6.0.2-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+        sha256: 80bab7bfc629882493af4aa31a4cfa43a4c57c83813253626916b8c7ada83476
+        only-arches:
+          - x86_64
+        x-checker-data:
+          type: pypi
+          name: PyYAML
+          packagetype: bdist_wheel
+
+  - name: vte
+    buildsystem: meson
+    sources:
+      - type: git
+        url: https://gitlab.gnome.org/GNOME/vte.git/
+        commit: 3f66edbf598129bafde3baa91ccfb345056418c3
+        
+  - name: WineCharm
+    buildsystem: simple
+    build-commands:
+      - pip3 install --no-deps --no-build-isolation --ignore-installed --prefix=${FLATPAK_DEST} .
+    sources:
+      - type: git
+        tag: 0.99.3
+        commit: e8a25fdb23199c0be99515e7f9230c06612c3390
+        url: https://github.com/fastrizwaan/WineCharm.git
```

Comment:
The main source code file is still not split into sensible parts, it is still the same 11k LOC file that makes it hard to review what the program does and be confident when accepting it.

this would be blocked on https://docs.flathub.org/docs/for-app-authors/requirements#structure-and-organisation

Reply:
Let me rephrase, how many files will make it good? Cause a single file already does what it is supposed to do. So, i'm asking you how many files would you like to see to be approving, i'll split them into that many.

Reply:
Now it is modular.